### PR TITLE
[ContentFragment] Add content fragment list component (#340)

### PR DESCRIPTION
**Pull request by philipmundt**
This PR is based on PRs https://github.com/adobe/aem-core-wcm-components/pull/499 and https://github.com/adobe/aem-core-wcm-components/pull/498

It adds/fixes feature #340 

Todos:
- [x] Merge https://github.com/adobe/aem-core-wcm-components/pull/498
- [x] Merge https://github.com/adobe/aem-core-wcm-components/pull/499
- [x] Rebase after integration of https://github.com/adobe/aem-core-wcm-components/pull/498 and https://github.com/adobe/aem-core-wcm-components/pull/499

### CODE REVIEW

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/servlets/contentfragment/AbstractContentFragmentDataSourceServlet.java [hunk 1]
```diff
@@ -26,60 +25,38 @@
...
 
 import static com.adobe.cq.wcm.core.components.models.contentfragment.ContentFragment.PN_PATH;
-import static org.apache.sling.api.resource.Resource.RESOURCE_TYPE_NON_EXISTING;
+
 
 /**
  * <p>Abstract datasource providing access to a content fragment instance configured by datasource properties. The
  * content fragment to be returned by {@link #getContentFragment(SlingHttpServletRequest)} can be specified via a
- * {@code /apps/core/wcm/components/contentfragment} component (see {@link #PN_COMPONENT_PATH}) or
+ * {@code /apps/core/wcm/extension/components/contentfragment} component (see {@link #PN_COMPONENT_PATH}) or
```
Comment (jckautzmann) [inline bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/servlets/contentfragment/AbstractContentFragmentDataSourceServlet.java]: Shouldn't `extension` be removed from the path?

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/contentfragment/ContentFragmentListImpl.java [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+            if (leakingResourceResolver != null) {
+                // Always close the leaking query builder resource resolver
+                leakingResourceResolver.close();
+            }
+        }
+    }
+
+    @NotNull
+    @Override
+    public Collection<DAMContentFragment> getListItems() {
```
Comment (jckautzmann) [inline bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/contentfragment/ContentFragmentListImpl.java]: I would have expected the items to be of type `ContentFragment` (content fragment component model).

File: content/src/content/jcr_root/apps/core/wcm/components/contentfragmentlist/v1/contentfragmentlist/clientlibs/dialog/js.txt [hunk 3]
```diff
@@ -0,0 +1,18 @@
+###############################################################################
+# Copyright 2018 Adobe Systems Incorporated
```
Comment (jckautzmann) [inline content/src/content/jcr_root/apps/core/wcm/components/contentfragmentlist/v1/contentfragmentlist/clientlibs/dialog/js.txt]: Year should be 2019.

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/contentfragment/ContentFragmentListImpl.java [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+        }
+
+        QueryBuilder queryBuilder = resourceResolver.adaptTo(QueryBuilder.class);
+        if (queryBuilder == null) {
+            LOG.warn("Query builder was null therefore no query was executed");
+            return;
+        }
+
+        Map<String, String> queryParameterMap = new HashMap<>();
+        queryParameterMap.put("path", parentPath);
```
Comment (abhishekagg19) [inline bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/contentfragment/ContentFragmentListImpl.java]: Do we need to add p.limit = -1 to the queryParameterMap?

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/servlets/contentfragment/AbstractContentFragmentDataSourceServlet.java (hunk 1 above)
Comment (philipmundt) [reply bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/servlets/contentfragment/AbstractContentFragmentDataSourceServlet.java]: Probably a merge error. Will fix this.

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/contentfragment/ContentFragmentListImpl.java (hunk 2 above)
Comment (philipmundt) [reply bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/contentfragment/ContentFragmentListImpl.java]: No, that would be wrong. Both models only share the `DAMContentFragment`.

File: content/src/content/jcr_root/apps/core/wcm/components/contentfragmentlist/v1/contentfragmentlist/clientlibs/dialog/js.txt (hunk 3 above)
Comment (philipmundt) [reply content/src/content/jcr_root/apps/core/wcm/components/contentfragmentlist/v1/contentfragmentlist/clientlibs/dialog/js.txt]: Will fix.

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/contentfragment/ContentFragmentListImpl.java (hunk 4 above)
Comment (philipmundt) [reply bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/contentfragment/ContentFragmentListImpl.java]: Could you please elaborate?

**Pull request merged**